kratos/includes/model_part.h

```diff
@@ -1988,6 +1955,190 @@ class KRATOS_API(KRATOS_CORE) ModelPart final
+     *            to add the items in the range.
+     *                  -- This will keep on increasing the capacity of the current PVS as well as all the
+     *                     parent PVS containers.
+     *
+     *  - In the case 2a:
+     *          - If it defines a subrange, we don't have to do anything. We can simply skip the insertion on the current PVS as
+     *            well as in all the parents because they must already have the subrange.
+     *                  -- This will keep the same capacity of the PVS.
+     *
+     *  - In the case 2b:
+     *          - If does not define a subrange and it is of type PVS, hence they are sorted and made unique. So we don't need
+     *            an auxiliary here as well.
+     *                  -- This will keep on increasing the capacity of the current PVS as well as all the
+     *                     parent PVS containers.
+     *
+     * @tparam TContainerGetter
+     * @tparam TIterator
+     * @param rContainerGetter
+     * @param begin
+     * @param end
+     */
+    template<class TContainerGetter, class TIterator>
+    void InsertEntityRange(
+        const TContainerGetter& rContainerGetter,
+        TIterator begin,
+        TIterator end)
+    {
+        KRATOS_TRY
+
+
+        ModelPart* p_current_part = this;
+
+        if constexpr(!std::is_same_v<iterator, std::remove_cv<TIterator>>) {
+            // this represents the case 1.
+            using container_type = std::remove_pointer_t<decltype(std::declval<TContainerGetter>()(std::declval<ModelPart*>()))>;
+            container_type aux;
+            aux.insert(begin, end);
+
+            // we add to all the parents and the current model part
+            while (p_current_part->IsSubModelPart()) {
+                rContainerGetter(p_current_part)->insert(aux.begin(), aux.end());
+                p_current_part = &(p_current_part->GetParentModelPart());
+            }
+
+            // now we add to the root model part
+            rContainerGetter(p_current_part)->insert(aux.begin(), aux.end());
+        } else {
+            // this represents the case 2.
+            // now the iterators belong to PVS::iterator, hence no sorting or making them to unique will occur.
+            while (p_current_part->IsSubModelPart() && !IsSubSet(*rContainerGetter(p_current_part), begin, end)) {
+                rContainerGetter(p_current_part)->insert(begin, end);
+                p_current_part = &(p_current_part->GetParentModelPart());
+            }
+
+            // now we do the same for the root model part
+            if (!IsSubSet(*rContainerGetter(p_current_part), begin, end)) {
+                rContainerGetter(p_current_part)->insert(begin, end);
+            }
+        }
+
+        KRATOS_CATCH("")
+    }
+
+    template<class TContainerGetter, class TInputContainer, std::enable_if_t<IsRValueContainer<TInputContainer>::value, bool> = true>
+    void InsertEntityRValuedContainer(
```

Comment:
no need to explain function arguments in the function's name
```suggestion
    void InsertEntities(
```

Reply:
done :)